packages/core/src/components/AutocompleteField/AutocompleteField.jsx

```diff
@@ -0,0 +1,175 @@
+import Button from '../Button/Button';
+import Downshift from 'downshift';
+import FormLabel from '../FormLabel/FormLabel';
+import PropTypes from 'prop-types';
+import React from 'react';
+import noop from 'nooop';
+import uniqueId from 'lodash.uniqueid';
+
+/**
+ * An `AutocompleteField` component renders a typeahead input field, form label, and a clear input link.
+ */
+export class AutocompleteField extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.id = uniqueId('autocomplete_');
+  }
+
+  render() {
+    const {
+      clearAriaLabel,
+      clearInputText,
+      disabled,
+      fieldRef,
+      items,
+      itemToString,
+      labelHint,
+      labelText,
+      onChange,
+      onStateChange,
+      ...fieldProps
+    } = this.props;
+
+    return (
+      <Downshift
+        itemToString={itemToString}
+        onChange={onChange}
+        onStateChange={onStateChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          inputValue
+        }) => (
+          <div className="ds-u-display--flex ds-u-flex-wrap--wrap ds-c-autocomplete">
+            <FormLabel
+              className="ds-u-margin-top--0"
+              labelClassName="ds-c-autocomplete__label"
+              component="label"
+              fieldId={this.id}
+              hint={labelHint}
+            >
+              {labelText}
+            </FormLabel>
+
+            <input
+              {...getInputProps({
+                className: 'ds-c-field ds-c-autocomplete__input',
+                disabled: disabled,
+                id: this.id,
+                ref: fieldRef,
+                ...fieldProps
+              })}
+            />
+
+            {isOpen ? (
+              <div className="ds-u-border--1 ds-u-padding--1 ds-c-autocomplete__list">
+                <ul className="ds-c-list--bare">
+                  {items
+                    .filter(
+                      item =>
+                        !inputValue ||
+                        item.name
+                          .toLowerCase()
+                          .includes(inputValue.toLowerCase())
+                    )
+                    .map((item, index) => (
+                      <li
+                        className={
+                          highlightedIndex === index
+                            ? 'ds-u-padding--1 ds-c-autocomplete__list-item ds-c-autocomplete__list-item--active'
+                            : 'ds-u-padding--1 ds-c-autocomplete__list-item'
+                        }
+                        key={item.id}
+                        {...getItemProps({ item })}
+                      >
+                        {itemToString(item)}
+                      </li>
+                    ))}
+                </ul>
+              </div>
+            ) : null}
+
+            <Button
```

Comment:
I feel a clear function is in order. Users should have a visual cue that this is not a standard input box. It would be a simple update to reset the `selectedItem` control prop on input focus, and maybe we should be doing that anyway. I still feel a secondary clear function is in order, otherwise we assume too much about users understanding this is a unique UI element.

Reply:
> It would be a simple update to reset the `selectedItem` control prop on input focus, and maybe we should be doing that anyway.

Do you mean on `blur`? I agree with that. We probably shouldn't rely on the user clearing the field by clicking the "Clear" button. Some users might erase the value and proceed with the rest of the form, so we wouldn't want a hidden value getting submitted to the backend in those scenarios.

> Users should have a visual cue that this is not a standard input box.

I think the autocomplete list that gets displayed is a pretty good visual cue already that the field has additional functionality, no? 

Your comment got me thinking though: There are two ways I can think of an autocomplete list being used, and I think one of these ways requires a stronger visual cue than the other:

1. A **constrained** autocomplete input, where the user *must* select from a constrained list of options. An example of a constrained autocomplete input is one where a user is asked to enter their zip code and they're required to select their county from the autocomplete list. For this category of autocomplete, it's really important that we make it clear to the user that they need to pick an option from the list. 

    Here's one idea, where we could use a label within the autocomplete list to indicate to the user that they need to select from the list. The exact phrasing we use might need some finessing.
  ![image](https://user-images.githubusercontent.com/371943/33853863-892d9586-de8d-11e7-856c-40288d757636.png)
2. An **unconstrained** autocomplete input, where the autocomplete list is there to help the user fill out the field quicker, but whatever value they enter is submitted to the backend. An example of an unconstrained autocomplete is a search field, where an autocomplete list could surface common search queries, but the field accepts any search query the user types. For this category, the autocomplete functionality is an enhancement and I think the autocomplete list itself is enough of a visual cue that the field has additional functionality.
  ![image](https://user-images.githubusercontent.com/371943/33854016-20ef5d5a-de8e-11e7-9050-658dabbbb862.png)

Curious to hear @cindyphan's thoughts on this 👂

Reply:
My example component is definitely a constrained one, but I envision this being used in an unconstrained way, with asynchronous data calls returning 0 to N results on the fly. When users select an option from the suggested list, the input should remember and retain that choice until the user opts to change it. This could be initiated by clicking the "Clear typeahead" link, or focusing the input again.

If my option were cleared out of the input field on blur, I would feel something was broken, and try searching again.

Reply:
I think there is a use for both of those autocompletes, things like ZIP Code search used in WindowShop/Screeners/FLH would need a constrained one, and I think adding the label "Select which county" or some other phrasing helps to clarify what needs to happen since we are using the autocomplete as the selection point. I do think we need some kind of clear button, "Search again" or "Clear". We could do the micro-blue-outline for it. 

Then the unconstrained is useful for more general searches, like help. (This is something we don't need right away however.)

I think something we'll need to account for as well is:
1. I typed in a ZIP Code - 60647
2. Selected a county from the dropdown list - Chicago, IL 60647
3. Purposely or mistakenly deleted from letters from my selection - Chicago, IL 6064
4. Now this selection is technically invalid

This is how Find Local Help is currently approaching it, will connect with Trevor about this more:
![image](https://user-images.githubusercontent.com/3190552/33893470-796f7d20-df29-11e7-9942-1243d6d3fbf0.png)

Reply:
I added a conditional Boolean for a constrained list header. By passing `constrainedList` to `Autocomplete`, you will see the header in the typeahead list. It has a prop for updating the header string as well.

I'll keep looking at Downshift to see if we can pass a conditional "No results found" message.